### DISCUSSION TITLE
License

### DISCUSSION THREAD

**Discussion by mautamu** [leftwm/leftwm#559] [Q&A]
Howdy,

While LeftWM itself is MIT licensed, LeftWM-Theme and LeftWM-community-themes are BSD licensed. Although, to the best of my knowledge, these two licenses are compatible, it may be prudent to relicense LeftWM-Theme and LeftWM-community-themes with the MIT license to be consistent throughout the LeftWM ecosystem.

I therefore propose the following options:

   1.  Leave LeftWM-Theme and LeftWM-community-themes as BSD, and move LeftWM to BSD
   2.  Leave LeftWM as MIT, and move LeftWM-Theme and LeftWM-community-themes to MIT
   3.  Continue with the license discrepancy as-is
   4.  Move LeftWM, LeftWM-community-themes, and LeftWM-Theme to the MPL 2.0, as it is common in the Rust ecosystem

I personally prefer options 2 or 4, as they will allow LeftWM to move closer to being a library upon which many LeftWM flavors might be built while also providing some consistency between the different projects within LeftWM.

I suppose I also should ask why we have the two different licenses to start with, as this change does affect past decisions as well.

Thoughts? (taken from https://github.com/leftwm/leftwm-theme/issues/30)

We probably should decide this prior to releasing leftwm-theme on crates.io or Copr.

Best,
Mautamu

**Comment by ghost** [comment]
The BSD-3-clause license sounds good, just as a small question :thinking: why aren't you considering the (A/L)GPL-2/3 ? Similar to the MPL it would help to prevent proprietary forks of leftwm and ensure that all derivatives stay libre.

**Reply by lex148** [reply]
I can speak to why it is the way it is currently, but if We as a community want to change it I would be happy to support that decision. 

I started with LeftWM and wanted to pick a license that is super flexible. Being a full time developer I have had the experience many times of wanting to lookup how to do some specific task. If you find an answer in another project somewhere you should make sure the license of that project permits you to reuse the code. This isn't always possible in your day job if you are dealing with a GPL code base. I truly respect the goals of the GPL and think it is the absolute best choice in a lot of cases. However for this specific project, I feel it is more important to share knowledge with everyone than it is to prevent proprietary forks. Lets be realistic here. If Google wants to make a window manager, they aren't going to be looking up LeftWM ㋡ . 

When I started the theme project I wanted to pick a license that was best for the community as a whole. Specifically I wanted to encourage people to fork themes like crazy, play and have fun. Again the `MIT` license looked like a good pick, however I was advised by some colleagues the `BSD-3-clause` might be a better pick. The argument I was presented was they do almost the exact same thing, but `BSD-3-clause` is legally worded better. Additionally `BSD-3-clause` prevents the endorsing of contributors, and this might be desirable to some people that might want to make themes. 

All that being said, LeftWM has grown just beyond my side project. I might be using it day to day, but many others are too. And many other contribute there hard work as well. I fully support where the community wants to go.

**Reply by AethanFoot** [reply]
Very well said @lex148. The points said is why BSD and MPL seem the most reasonable as BSD (as you said) is basically at the same level as MIT but with a few niceties. If copy-left is important to some people, MPL would be best as anyone can use it (no matter the projects overall license) even in proprietary software, just the files with the MPL'd code stay MPL'd. Compared to GPL which requires the whole project to share the license.

I don't really have much of a preference myself, however I would say at this point BSD would be better (to synchronise the licenses)  as it allows us to re-evaluate in the future if (or when) the project becomes much bigger.

**Reply by ghost** [reply]
For the looks of these replies and after investigating a bit on my own, the MPL sounds like a good option, since any proprietary additions can stay proprietary **BUT** if any improvements are made on the core those must be shared with the devs. If there is no real reason to keep the core leftwm code that protected it would be wise to go with BSD-3-clause instead :D
